core/slice.go

```diff
@@ -161,59 +167,107 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 		newInboundEtxs, subRollup, err = sl.CollectNewlyConfirmedEtxs(block, block.Location())
 		if err != nil {
 			log.Debug("Error collecting newly confirmed etxs: ", "err", err)
-			return nil, ErrSubNotSyncedToDom
+			return nil, nil, nil, false, ErrSubNotSyncedToDom
 		}
 	} else if nodeCtx < common.ZONE_CTX {
 		subRollups, err := sl.CollectSubRollups(block)
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, false, err
 		}
 		subRollup = subRollups[nodeCtx+1]
 	}
 
 	// Append the new block
 	err = sl.hc.Append(batch, block, newInboundEtxs.FilterToLocation(common.NodeLocation))
 	if err != nil {
-		return nil, err
-	}
-
-	// CalcTd on the new block
-	td, err := sl.calcTd(block.Header(), domTd, domOrigin)
-	if err != nil {
-		return nil, err
-	}
-	if !domOrigin {
-		// HLCR
-		reorg = sl.hlcr(td)
+		return nil, nil, nil, false, err
 	}
 
 	// Upate the local pending header
 	localPendingHeader, err := sl.miner.worker.GeneratePendingHeader(block)
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, false, err
 	}
 
 	// Combine subordinates pending header with local pending header
 	pendingHeaderWithTermini := sl.computePendingHeader(types.PendingHeader{Header: localPendingHeader, Termini: newTermini}, domPendingHeader, domOrigin)
 
+	// Find parent subentropy
+	parentDeltaS := sl.hc.GetDeltaS(header.ParentHash(), header.NumberU64()-1)
+
+	var subDeltaS *big.Float
+	// Make new deltaS array
+	newDeltaS := make([]*big.Float, 4)
+	for i, deltaS := range parentDeltaS {
+		newDeltaS[i] = deltaS
+	}
+	var s *big.Float
+	parentS := sl.hc.GetS(header.ParentHash(), header.NumberU64()-1)
+	if nodeCtx != common.ZONE_CTX {
```

Comment:
we  probably don't need 2 filters here for nodeCtx since this logic only applies to region. just `if nodeCtx == common.REGION_CTX` would suffice.